src/Parser/RetributionPaladin/Modules/PaladinCore/BlessingOfTheAshbringer.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Module from 'Parser/Core/Module';
+
+const debug = false;
+
+class BlessingOfTheAshbringer extends Module {
+	static dependencies = {
+		combatants: Combatants,
+	};
+
+
+	//Blessing of the Ashbringer = BotA
+	botaUptime = 0;
+	blessingOfWisdomUptime = 0;
+	blessingOfKingsUptime = 0;
+
+	on_initialized() {
+		this.active = this.combatants.selected.traitsBySpellId[SPELLS.BLESSING_OF_THE_ASHBRINGER.id] > 0;
+	}
+
+
+	on_byPlayer_applybuff(event) {
+		const spellId = event.ability.guid;
+		const targetId = event.targetID;
+		this.botaUptime = this.combatants.selected.getBuffUptime(SPELLS.BLESSING_OF_THE_ASHBRINGER_BUFF.id) / this.owner.fightDuration;
```

Comment:
This would be better as a getter. I fear if this event isn't called in a long time the value will be off.

```jsx
get uptime() {
  return this.combatants.selected.getBuffUptime(SPELLS.BLESSING_OF_THE_ASHBRINGER_BUFF.id) / this.owner.fightDuration;
}
```